Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp

```diff
@@ -449,23 +448,38 @@ void WebResourceLoadStatisticsStore::setLoginStatus(RegistrableDomain&& domain,
 {
     ASSERT(RunLoop::isMain());
 
-    if (loggedInStatus == IsLoggedIn::LoggedIn) {
-        auto loginStatusToSet = lastAuthentication && lastAuthentication->hasExpired() ? std::nullopt : std::optional(WTFMove(lastAuthentication));
-        if (loginStatusToSet)
-            loginStatusToSet->setTimeToLive(WebCore::LoginStatus::TimeToLiveLong);
-        auto pair = std::make_pair(loggedInStatus, loginStatusToSet);
-        m_loginStatus.set(domain, WTFMove(pair));
-    } else
-        m_loginStatus.remove(domain);
-    completionHandler();
+    auto loginStatusToSet = lastAuthentication && lastAuthentication->hasExpired() ? std::nullopt : std::optional(WTFMove(lastAuthentication));
+    if (loginStatusToSet)
+        loginStatusToSet->setTimeToLive(WebCore::LoginStatus::TimeToLiveLong);
+
+    postTask([this, domain = WTFMove(domain), loggedInStatus, lastAuthentication = WTFMove(lastAuthentication), completionHandler = WTFMove(completionHandler)]() mutable {
```

Comment:
That's a bug. postTask() dispatches to another thread so you need `crossThreadCopy()` the non-thread safe types you're capturing in the lambda:
```
domain = crossThreadCopy(WTFMove(domain)), lastAuthentication = crossTheadCopy(WTFMove(lastAuthentication))
```

Note that this will require adding an `isolatedCopy()` member function to `LoginStatus`.